proxy/src/http/project.rs

```diff
@@ -11,26 +11,40 @@ use crate::http;
 use crate::project;
 use crate::registry;
 
-/// Combination of all routes.
+/// Prefixed filters.
+pub fn routes<R>(ctx: http::Ctx<R>) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone
+where
+    R: registry::Client + 'static,
+{
+    path("projects").and(
+        get_filter(ctx.clone())
+            .or(create_filter(ctx.clone()))
+            .or(discover_filter(ctx.clone()))
+            .or(list_filter(ctx)),
+    )
+}
+
+/// Combination of all project filters.
+#[cfg(test)]
```

Comment:
What does this do?

Reply:
The [`cfg`](https://doc.rust-lang.org/stable/rust-by-example/attribute/cfg.html) attribute helps to conditionally compile code. In this case it guards this function to be only available in the context of tests. The reason for this indirection is that we in our module tests when we assert endpoint behaviour are not interested in the routing prefix/namespace. This oddity will be addressed in a small refactor of the http stack.